platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -587,7 +598,12 @@ OtaPalStatus_t otaPal_SetPlatformImageState( OtaFileContext_t * const C,
     if( ( eState != OtaImageStateUnknown ) && ( eState <= OtaLastImageState ) )
     {
         /* Get file path for the image state file. */
-        getFilePathFromCWD( imageStateFile, OTA_PLATFORM_IMAGE_STATE_FILE );
+        status = getFilePathFromCWD( imageStateFile, OTA_PLATFORM_IMAGE_STATE_FILE );
+
+        if( status != OtaPalFileGenSuccess )
+        {
+            LogError( ( "Could not generate the absolute path for the file" ) );
+        }
 
         /* POSIX port using standard library */
         /* coverity[misra_c_2012_rule_21_6_violation] */
```

Comment:
If `status != OtaPalFileGenSuccess` on line 603, you're opening an invalid filename here.  You should not continue processing without a valid filename.

Reply:
Refactored in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1487/commits/a3d852db7cfbb536eddf4de696d8ec0f070a7702